lib/proto.c

```diff
@@ -893,6 +934,20 @@ void quicrq_wakeup_media_stream(quicrq_stream_ctx_t* stream_ctx)
                 (stream_ctx->is_cache_real_time && !stream_ctx->is_cache_policy_sent)) {
                 picoquic_mark_active_stream(stream_ctx->cnx_ctx->cnx, stream_ctx->stream_id, 1, stream_ctx);
             }
+        } else if (stream_ctx->transport_mode == quicrq_transport_mode_warp) {
+            quicrq_uni_stream_ctx_t* uni_stream_ctx = quicrq_find_uni_stream_for_group(stream_ctx, stream_ctx->media_ctx->current_group_id);
+            if (uni_stream_ctx == NULL && stream_ctx->media_ctx->current_group_id == stream_ctx->start_group_id) {
+                /* first ever warp stream */
+                uint64_t stream_id = picoquic_get_next_local_stream_id(stream_ctx->cnx_ctx, 1);
+                quicrq_find_or_create_uni_stream(stream_ctx->cnx_ctx, stream_id, 1);
+
+            } else {
+
+            }
+            // if in the same group-id , picoquic_mark_active_stream(for the stream_id mapping to the group-id)
```

Comment:
The previous else block is not needed. You just need to check `uni_stream_ctx` again, and activate the uni_stream so it will be polled by the picoquic callback function.

Reply:
IIUC , following 3 cases are needed
1. if the current group is still active ==> just mark the stream active
2. if its a new group 
     create a new stream and mark it active 
     if not first start group  ==> stop the stream with current_group_id - 1
     
Am i correct in understanding ?